src/aws.ts

```diff
@@ -440,15 +440,17 @@ export namespace $AWS {
         if (payload === undefined) {
           throw new Error("missing property 'payload'");
         }
-        return {
-          Type: "Task",
-          Resource: "arn:aws:states:::lambda:invoke",
-          Parameters: {
-            FunctionName: functionRef.resource.functionName,
-            [`Payload${payload && isVariableReference(payload) ? ".$" : ""}`]:
-              payload ? context.toJson(payload) : null,
-          },
-        };
+
+        return context.evalExpr(payload, (output) => {
+          return context.outputState({
+            Type: "Task",
+            Resource: "arn:aws:states:::lambda:invoke",
+            Parameters: {
+              FunctionName: functionRef.resource.functionName,
+              ...context.toJsonAssignment("Payload", output),
```

Comment:
It optionally adds the `.$` based on the output being a `JsonPath` or a `Value` (I added this method a few months ago, just changed it to support `ASLGraph.Output` instead of Expr.